secboot/secboot_tpm.go

```diff
@@ -377,18 +374,29 @@ func SealKey(key EncryptionKey, params *SealKeyParams) error {
 		return err
 	}
 
-	// Provision the TPM as late as possible
-	if err := tpmProvision(tpm, params.TPMLockoutAuthFile); err != nil {
-		return err
+	if params.TPMProvision {
+		// Provision the TPM as late as possible
+		if err := tpmProvision(tpm, params.TPMLockoutAuthFile); err != nil {
+			return err
+		}
 	}
 
-	// Seal key to the TPM
+	// Seal the provided keys to the TPM
 	creationParams := sb.KeyCreationParams{
 		PCRProfile:             pcrProfile,
-		PCRPolicyCounterHandle: policyCounterHandle,
+		PCRPolicyCounterHandle: tpm2.Handle(params.PCRPolicyCounterHandle),
+		AuthKey:                params.TPMPolicyAuthKey,
+	}
+
+	sbKeys := make([]*sb.SealKeyRequest, 0, len(keys))
+	for i := range keys {
+		sbKeys = append(sbKeys, &sb.SealKeyRequest{
+			Key:  keys[i].Key[:],
+			Path: keys[i].KeyFile,
+		})
 	}
 
-	authKey, err := sbSealKeyToTPM(tpm, key[:], params.KeyFile, &creationParams)
+	authKey, err := sbSealKeyToTPMMultiple(tpm, sbKeys, &creationParams)
 	if err != nil {
 		return err
 	}
```

Comment:
the write to the fail after this should be optional if AuthKey was passed in and the file name is empty